### ISSUE TITLE
SystemExit and being unable to set live-py variables

### ISSUE THREAD

**Issue by RidaAyed** [donkirkby/live-py-plugin#643]
I'm somehow unable to `live-py` with unittests.  
```emacs-lisp
(setq default-directory "/home/ra/src/app")
(setq live-py-dir "/home/ra/src/app")
(setq live-py-version "/home/ra/.local/share/virtualenvs/app-Rqr0N7FF/bin/python3")
(setq live-py-driver "-m pytest -k foobar")
```

-   Now open I open `/home/ra/src/app/test_foo.py`
-   Start live-py
-   SystemExit: ExitCode.USAGE_ERROR | pytest doesn't call test_foo.py. Try a different driver.

Looking at the variables they're reset  

```emacs-lisp
(describe-variable 'live-py-dir) ;; Is reset: /home/ra/src/app/tests/
(describe-variable 'live-py-version) ;; Is reset to: python
(describe-variable 'live-py-driver) ;; Remains as set
```
Furthermore setting either `live-py-version` or `live-py-dir`, manually from the python buffer, immediately starts live-py.

**Comment by donkirkby** [comment]
I guess I never considered letting users set those variables before starting Live Python mode. Instead, I set them to reasonable defaults when you start the mode, and then mapped keys to set them after starting the mode.

```
            (define-key map (kbd "C-c M-d") #'live-py-set-driver)
            (define-key map (kbd "C-c M-w") #'live-py-set-dir)
            (define-key map (kbd "C-c M-p") #'live-py-set-path)
            (define-key map (kbd "C-c M-v") #'live-py-set-version)
            (define-key map (kbd "C-c M-a") #'live-py-set-args)
```

One trick I use is to activate a Python virtual environment before launching emacs. That way, the `python` default works and I don't have to set it.

**Comment by donkirkby** [comment]
I'm going to close this issue, since it's working as designed.

**Comment by RidaAyed** [comment]
tldr; How can I sove this issue?
Thank you for taking your time, I appreciate it. My venvs are (automatically) launched on entering their directories (using envrc).  
Apparently the variables `live-py-dir` and `version` are not set or at least reset to wrong values (as shown in the subsequent comments above).
What must I do in order to set these variables?

**Comment by donkirkby** [comment]
You can use the mapped keys I listed above, after you start Live Python mode. The equivalent to what you posted in the question would be:

```
M-x live-py-mode
C-c M-w /home/ra/src/app
C-c M-v /home/ra/.local/share/virtualenvs/app-Rqr0N7FF/bin/python3
C-c M-d -m pytest -k foobar
```

If your virtualenv is active when you launch emacs, then `C-c M-v` shouldn't be necessary.

**Comment by RidaAyed** [comment]
Of course, now the penny has droped. Thank you for your patience :clap:
